src/Datadog.Trace.ClrProfiler.Managed/Integrations/HttpMessageHandlerIntegration.cs

```diff
@@ -231,8 +254,10 @@ public static class HttpMessageHandlerIntegration
                         SpanContextPropagator.Instance.Inject(scope.Span.Context, new ReflectionHttpHeadersCollection(headers));
                     }
 
-                    var task = (Task<T>)sendAsync(handler, request, cancellationToken);
-                    var response = await task.ConfigureAwait(false);
+                    var task = (Task)sendAsync(handler, request, cancellationToken);
+                    await task.ConfigureAwait(false);
+
+                    var response = task.GetProperty("Result").Value;
```

Comment:
Maybe we can avoid the Lookup (and maybe allocations) inside GetProperty here by declaring a static var with a copy of the delegate compiled from the expression tree to extract the result. If we already know what is the type of T, like `HttpResponseMessage` we can pass the delegate as a parameter to SendAsyncInternal. So the caller pass the delegate to extract the value of the task.

Reply:
`GetProperty` does not allocate. You're right that it would save a lookup, but I think the overhead is negligible with the current state of the method (see the measured overhead of GetProperty here: https://github.com/DataDog/dd-trace-dotnet/pull/759). I think that's something we should keep for another round of optimizations.

Reply:
Ok 👍